prog/github/github_repository_nexus.rb

```diff
@@ -62,10 +70,11 @@ def before_run
   end
 
   label def wait
+    polling_interval = 5 * 60
     should_destroy = (Time.now - github_repository.last_job_at > 6 * 60 * 60)
 
     begin
-      check_queued_jobs
+      polling_interval = check_queued_jobs
```

Comment:
You asked about what kind of design can we use to not add a secondary purpose (returning polling_interval) to check_queued_jobs.

How about moving processing remain quota out of check_queued_jobs. Something like this;

```ruby
...
begin
  remaining_quota_percentage = client.rate_limit.remaining / client.rate_limit.limit.to_f
  check_queued_jobs if remaining_quota_percentage >= 0.1
  polling_interval = <calculate-based-on-remaining-quota-percentage>
  nap polling_interval 
rescue
 ...
end
...
```

Reply:
When we move it out from `check_queued_jobs`, `client.rate_limit` generates an extra API request. In other words, even if we don't make any API calls for that client, it needs to fetch the rate limits from GitHub.

I changed it. Since `check_queued_jobs` isn't a separate label, it put the polling interval to `@polling_interval` variable.

Reply:
Yep, but don't we call `rate_limit` in all cases in `check_queued_jobs`? This just changes the place we call it, not the times we call. Did I miss anything?

Reply:
If the same client object has a previous API call, it uses the `last_response` instead of an extra API call.
https://github.com/octokit/octokit.rb/blob/ceea40bb999ef868f6f1ab25336518127a17c3ee/lib/octokit/client/rate_limit.rb#L14